src/App/DataSource/DataSourceManager.php

```diff
@@ -47,55 +51,89 @@ class DataSourceManager
      * @param  Laravel\Lumen\Routing\Router  $router
      * @return void
      */
-    public function __construct(\Laravel\Lumen\Routing\Router $router)
+    public function __construct(ConfigRepository $configRepo)
     {
-        $this->router = $router;
+        $this->configRepo = $configRepo;
     }
 
-    public function addSource(DataSource $source)
+    public function getSources() : array
     {
-        $this->sources[$source->getId()] = $source;
+        return array_keys($this->sources);
     }
 
-    public function getSource($name = false)
+    /**
+     * Get an array of enabled source names
+     *
+     * @return [string, ...]
+     */
+    public function getEnabledSources() : array
     {
-        if ($name) {
-            return isset($this->sources[$name]) ? $this->sources[$name] : false;
-        }
+        // Load enabled sources
+        $enabledSources = array_filter(
```

Comment:
This is a bit gross since it makes DataSourceManager reliant on the config repo *but* it seems better than hiding that detail in a service provider instead